src/layout-delay-meter.js

```diff
@@ -44,6 +51,29 @@ export class LayoutDelayMeter {
     this.done_ = false;
     /** @private {?TickLabel} */
     this.label_ = LABEL_MAP[priority];
+
+    this.initViewportObserver_();
+  }
+
+  /**
+   * Initializes viewport observer for calling `enterViewport`.
+   * @private
+   */
+  initViewportObserver_() {
+    if (!viewportObservers.has(this.win_)) {
+      viewportObservers.set(
+        this.win_,
+        createViewportObserver((entries) => {
+          for (let i = 0; i < entries.length; i++) {
+            const {target, isIntersecting} = entries[i];
+            if (isIntersecting) {
+              target.getLayoutDelayMeter().enterViewport();
+            }
+          }
+        }, this.win_)
+      );
+    }
+    viewportObservers.get(this.win_).observe(this.element_);
```

Comment:
One thing that's missing is the `unobserve` when the element is disconnected. Overall the modal of the layout-delay-meter is super similar to the loading indicator. But I refactored it slightly different. And IMHO that looked a bit better. In that model this would look like this:

1. There's a single "instance" per window that deals with layout-delay-meter for any element in that window. Let's call it `LoadDelayTracker`.
2. CE only needs to call `track(this)` and `untrack(this)` on the `LoadDelayTracker`.
3. The `LoadDelayTracker` would instantiate a single instance of InOb for its window.
4. The `LoadDelayTracker` would keep some minimal per-element state in a `WeakMap`.
5. That's it.

Reply:
Great idea. Its a bigger shift away in code style / a bigger refactor, but I agree it'd be much cleaner! Luckily we can just remove it outright :)